### PR TITLE
community: added Browserbase loader

### PR DESCRIPTION
**Pull request by mishushakov**
Thank you for contributing to LangChain!

- [x] **PR title**: "community: added Browserbase loader"
  - Where "package" is whichever of langchain, community, core, experimental, etc. is being modified. Use "docs: ..." for purely docs changes, "templates: ..." for template changes, "infra: ..." for CI changes.
  - Example: "community: add foobar LLM"

- [x] **PR message**: ***Delete this entire checklist*** and replace with
    - **Description:** Adds remote browser web-page loader powered by browserbase.com
    - **Issue:** -
    - **Dependencies:** (optional) browserbase
    - **Twitter handle:** @browserbasehq

- [x] **Add tests and docs**: If you're adding a new integration, please include
  1. a test for the integration, preferably unit tests that do not rely on network access,
  2. an example notebook showing its use. It lives in `docs/docs/integrations` directory.

- [x] **Lint and test**: Run `make format`, `make lint` and `make test` from the root of the package(s) you've modified. See contribution guidelines for more: https://python.langchain.com/docs/contributing/

Additional guidelines:
- Make sure optional dependencies are imported within a function.
- Please do not add dependencies to pyproject.toml files (even optional ones) unless they are required for unit tests.
- Most PRs should not touch more than one package.
- Changes should be backwards compatible.
- If you are adding something to community, do not re-import it in langchain.

If no one reviews your PR within a few days, please @-mention one of baskaryan, efriis, eyurtsev, hwchase17.

### CODE REVIEW

File: libs/langchain/langchain/document_loaders/browserbase.py [hunk 1]
```diff
@@ -0,0 +1,3 @@
+from langchain_community.document_loaders.browserbase import BrowserbaseLoader
```
Comment (eyurtsev) [inline libs/langchain/langchain/document_loaders/browserbase.py:1]: Could you remove this -- we're about to break dependency of langchain on langchain_community

File: libs/langchain/langchain/document_loaders/__init__.py [hunk 2]
```diff
@@ -85,6 +85,7 @@ def __getattr__(name: str) -> Any:
     "BlobLoader",
     "BlockchainDocumentLoader",
     "BraveSearchLoader",
+    "BrowserbaseLoader",
```
Comment (eyurtsev) [inline libs/langchain/langchain/document_loaders/__init__.py:88]:
```suggestion
```

File: libs/langchain/tests/unit_tests/document_loaders/test_imports.py [hunk 3]
```diff
@@ -35,6 +35,7 @@
     "BlobLoader",
     "BlockchainDocumentLoader",
     "BraveSearchLoader",
+    "BrowserbaseLoader",
```
Comment (eyurtsev) [inline libs/langchain/tests/unit_tests/document_loaders/test_imports.py:38]:
```suggestion
```

File: libs/community/langchain_community/document_loaders/browserbase.py [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+from langchain_community.document_loaders.base import BaseLoader
+
+
+class BrowserbaseLoader(BaseLoader):
+    """Create new Browserbase loader"""
+
+    def __init__(
+        self,
+        urls: List[str],
```
Comment (eyurtsev) [inline libs/community/langchain_community/document_loaders/browserbase.py:13]:
```suggestion
        urls: Sequence[str],
        *,
```

File: libs/community/langchain_community/document_loaders/browserbase.py [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+        api_key: str = os.environ["BROWSERBASE_KEY"],
```
Comment (eyurtsev) [inline libs/community/langchain_community/document_loaders/browserbase.py:14]: See how other integrations pick up information from the enviornment. We generally do not want code accessing the environment during initialization time.

```
        values["api_key"] = get_from_dict_or_env(
            values, "api_key", "BROWSERBASE_API_KEY"
        )

```

File: libs/community/langchain_community/document_loaders/browserbase.py [hunk 6]
```diff
@@ -0,0 +1,43 @@
+import os
+from typing import Iterator, List
```
Comment (eyurtsev) [inline libs/community/langchain_community/document_loaders/browserbase.py:2]:
```suggestion
from typing import Iterator, List, Sequence
```

File: libs/community/langchain_community/document_loaders/browserbase.py [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+from langchain_core.documents import Document
+
+from langchain_community.document_loaders.base import BaseLoader
+
+
+class BrowserbaseLoader(BaseLoader):
```
Comment (eyurtsev) [inline libs/community/langchain_community/document_loaders/browserbase.py:8]: Could you add a link here for users to explain what Browser Base is? What does this do? Why not just use requests? etc

File: docs/api_reference/guide_imports.json [hunk 8]
```diff
@@ -1 +1 @@
...
...al_qa", "QA using Activeloop's DeepLake": "https://python.langchain.com/docs/use_cases/question_answering/integrations/semantic-search-over-chat", "Neptune Open Cypher QA Chain": "https://python.langchain.com/docs/use_cases/more/graph/neptune_cypher_qa", "NebulaGraphQAChain": "https://python.langchain.com/docs/use_cases/more/graph/graph_nebula_qa", "Memgraph QA chain": "https://python.langchain.com/docs/use_cases/more/graph/graph_memgraph_qa", "KuzuQAChain": "https://python.langchain.com/docs/use_cases/more/graph/graph_kuzu_qa", "HugeGraph QA Chain": "https://python.langchain.com/docs/use_cases/more/graph/graph_hugegraph_qa", "GraphSparqlQAChain": "https://python.langchain.com/docs/use_cases/more/graph/graph_sparql_qa", "Ontotext GraphDB QA Chain": "https://python.langchain.com/docs/use_cases/more/graph/graph_ontotext_graphdb_qa", "Diffbot Graph Transformer": "https://python.langchain.com/docs/use_cases/more/graph/diffbot_graphtransformer", "ArangoDB QA chain": "https://python.langchain.com/docs/use_cases/more/graph/graph_arangodb_qa", "Neo4j DB QA chain": "https://python.langchain.com/docs/use_cases/more/graph/graph_cypher_qa", "FalkorDBQAChain": "https://python.langchain.com/docs
```
Comment (eyurtsev) [inline docs/api_reference/guide_imports.json:1]: Don't think there's a reason to modify this file?

**Review by eyurtsev — COMMENTED** [review]
Looks pretty neat!

**Comment by mishushakov** [comment]
Thank you so much! Ready for another round 😄

**Review by baskaryan — APPROVED** [review]

**Pull request merged**